path_service_principal.go

```diff
@@ -98,7 +110,7 @@ func (b *azureSecretBackend) pathSPRead(ctx context.Context, req *logical.Reques
 		"app_object_id":        role.ApplicationObjectID,
 		"key_id":               role.Credentials.KeyId,
 		"sp_object_id":         role.ServicePrincipalID,
-		"role_assignment_ids":  roleIDs(role.AzureRoles),
+		"role_assignment_ids":  raIDs,
```

Comment:
This was a bug introduced when I added the `token` endpoint. As a result if the `creds` endpoint is called (only a few things rely on this) then you could end up with bad role assignments in Azure which look like:

<img width="1226" alt="Screen Shot 2020-10-23 at 5 28 27 PM" src="https://user-images.githubusercontent.com/540634/97061742-2b6e8980-1555-11eb-96ee-7223abe9e3e6.png">

Reply:
Great catch.